packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -394,6 +394,52 @@ export default function parse(source: string, state: State): TemplateParseResult
         element.lwc = lwcOpts;
     }
 
+    function applyLwcRenderModeDirective(element: IRElement) {
+        const { parent, tag, attrsList } = element;
+
+        // Can't handle slots in applySlot because it would be too late for class and style attrs
+        if (state.renderMode === LWCDirectiveRenderMode.light && tag === 'slot') {
+            const invalidAttrs = attrsList
+                .filter(({ name }) => name !== 'name')
+                .map(({ name }) => name);
+            if (invalidAttrs.length > 0) {
+                return warnOnElement(
+                    ParserDiagnostics.LWC_LIGHT_SLOT_INVALID_ATTRIBUTES,
+                    element.__original,
+                    [invalidAttrs.join(',')]
+                );
+            }
+        }
```

Comment:
Technically `<slot>`s aren't allowed at all right now in light DOM mode, correct? (Until #2317 is merged.) But I understand proactively adding the attribute check.